### PR TITLE
Deprecate undesirable DateTime methods

### PR DESCRIPTION
**Pull request by daviditen**
We've decided to remove 'type datetime.today()', the "datetime - date" operator,
and 'type datetime.timeSinceEpoch():real'.

Equivalent replacements are: 'type datetime.now()', "datetime - new datetime(date, new time())", and
'timeSinceEpoch().totalSeconds(): timedelta' (a standalone function).

While there also deprecate/replace 'timedelta.total_seconds()'
with a camelCase version.

Added a new datetime initializer that takes a 'date' and 'time' argument. Added
a test of this initializer.

Added test/deprecated tests for everything that was deprecated.

Signed-off-by: David Iten <daviditen@users.noreply.github.com>

### CODE REVIEW

**Comment by daviditen** [comment]
Issue https://github.com/chapel-lang/chapel/issues/18834

File: test/deprecated/datetimeMinusDate.chpl [hunk 1]
```diff
@@ -0,0 +1,6 @@
+use DateTime;
+
+const dt = datetime.now(),
+      d  = date.today();
+
+const td = dt - d;
```
Comment (lydia-duncan) [inline test/deprecated/datetimeMinusDate.chpl:6]: I think it'd probably be good to ensure the intended behavior of the deprecated symbol is still maintained until it is removed, so would like to see `td` verified in some way (though I recognize it is difficult with these symbols, given their reliance on the current time)

**Review by lydia-duncan — APPROVED** [review]
Just a minor nit, otherwise I think this looks good!

File: test/deprecated/datetimeMinusDate.chpl (hunk 1 above)
Comment (daviditen) [reply test/deprecated/datetimeMinusDate.chpl:6]: Thanks!

I updated two of the four tests to add some functionality verification. The other two are less straightforward to verify though.

**Pull request merged**